modules/react-components/src/components/file-picker/file-picker.tsx

```diff
@@ -0,0 +1,685 @@
+    const [ hasError, setHasError ] = useState<boolean>(false);
+    const [ errorMessage, setErrorMessage ] = useState<string>(null);
+
+    // Hooks
+
+    useEffect(() => {
+        // Query the preferred color scheme.
+        const mql = window.matchMedia("(prefers-color-scheme:dark)");
+        // Check and set the dark mode initially.
+        if (mql?.matches) setDark(true);
+        // Callback for triggering the same for change events.
+        const triggerColorScheme = (event): void => {
+            setDark(event.matches ?? false);
+        };
+        // Check to see if match media API is available with the browser.
+        if (mql?.addEventListener) {
+            mql.addEventListener("change", triggerColorScheme);
+        }
+        // Cleanup logic.
+        return () => {
+            if (mql?.addEventListener) {
+                mql.removeEventListener("change", triggerColorScheme);
+            }
+        };
+    }, []);
+
+    useEffect(() => {
+        if (onChange) {
+            onChange({
+                file: file,
+                pastedContent: pastedContent,
+                serialized: serializedData,
+                valid: !hasError
+            });
+        }
+    }, [ file, pastedContent ]);
+
+    // Functional logic that used by event handlers to
+    // update the state of the picker.
+
+    const validate = async (data: File | string): Promise<boolean> => {
+        try {
+            await fileStrategy.validate(data); // Anonymous result
+            setHasError(false);
+            setErrorMessage(null);
+            return true;
+        } catch ({ errorMessage }) {
+            setHasError(true);
+            setErrorMessage(errorMessage);
+            return false;
+        }
+    };
+
+    // TODO: As a improvement implement the context component (this)
+    //       to support multiple file upload/attach strategy. If attaching
+    //       multiple then tabs Paste option should be removed dynamically.
+
+    // TODO: As a improvement compare newly added file or the
+    //       pasted content with the previous state and set it
+    //       to the state only if its a new change.
+
+    const addFileToState = async (file: File): Promise<void> => {
+        await validate(file);
+        setFile(file);
+        try {
+            setSerializedData(await fileStrategy.serialize(file));
+        } catch ({ errorMessage }) {
+            setHasError(true);
+            setErrorMessage(errorMessage);
+        }
+    };
+
+    const addPastedDataToState = async (text: string): Promise<void> => {
+        await validate(text);
+        setPastedContent(text);
+        try {
+            setSerializedData(await fileStrategy.serialize(text));
+        } catch ({ errorMessage }) {
+            setHasError(true);
+            setErrorMessage(errorMessage);
+        }
+    };
+
+    // Events that takes care of the drag drop input. User drops
```

Comment:
Let's remove these dangling comments and add doc comments instead.
We can do this in a subsequent PR.